src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopLibrary.cs

```diff
@@ -459,5 +459,39 @@ public void It_marks_extension_references_as_externally_resolved(bool? markAsExt
             frameworkReferenceCount.Should().BeGreaterThan(0);
             extensionReferenceCount.Should().BeGreaterThan(0);
         }
+
+        [FullMSBuildOnlyFact]
+        public void It_tolerates_newline_in_hint_path()
+        {
+            TestProject project = new TestProject()
+            {
+                Name = "NETFrameworkLibrary",
+                TargetFrameworks = "net462",
+                IsSdkProject = true
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(project, "SimpleNamesWithHintPathsWithNewLines")
+                .WithProjectChanges((path, p) =>
+                {
+                    if (Path.GetFileNameWithoutExtension(path) == project.Name)
```

Comment:
You shouldn't need to compare the project path unless you are creating multiple projects.  Since you only have one here you shouldn't need this.

Reply:
Fixed